File: Algorithms/Implementation/Breaking the Records.py
# Breaking the Records

# https://www.hackerrank.com/challenges/breaking-best-and-worst-records/problem

def breakingRecords(scores):
  # Write your code here
  minVal, maxVal = scores[0], scores[0]
  minCount, maxCount = 0, 0

  for i in range(1, len(scores)):
    if scores[i] < minVal:
      minCount += 1
      minVal = scores[i]
    if scores[i] > maxVal:
      maxCount += 1
      maxVal = scores[i]

  return [maxCount, minCount]

File: Algorithms/Implementation/Breaking the Records.js
// Breaking the Records

// https://www.hackerrank.com/challenges/breaking-best-and-worst-records/problem?isFullScreen=true

// #1
function breakingRecords(scores) {
  let min = scores[0]
  let max = min
  let countMin = 0
  let countMax = 0

  for (let i = 1; i < scores.length; i++) {
    if (scores[i] < min) {
      min = scores[i]
      countMin++
    }
    if (scores[i] > max) {
      max = scores[i]
      countMax++
    }
  }

  return [countMax, countMin]
}